lib/reek/core/smell_repository.rb

```diff
@@ -24,7 +24,9 @@ def initialize(source_description = nil, smell_types = self.class.smell_types)
       end
 
       def configure(klass, config)
-        @detectors[klass].configure_with(config) if @detectors[klass]
+        detector = @detectors[klass]
+        raise ArgumentError, "Cannot configure smell #{klass}" unless detector
```

Comment:
> `"Cannot configure nknown smell detector #{klass}"`

That still sounds confusing to me. How about `"Unknown smell type #{klass} found in confuguration"`?